File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int prog = arr[1] - arr[0];
        int n = arr.size();
        for (int i=2; i<n; i++) {
            if ((arr[i] - arr[i-1]) != prog) {
                return false;
            }
        }
        return true;
    }
};

File: 1502-can-make-arithmetic-progression-from-sequence/1502-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        arr.sort()
        prog = arr[1] - arr[0]
        n = len(arr)
        for i in range(2, n):
            if (arr[i] - arr[i-1]) != prog:
                return False
        return True
